fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -108,10 +112,32 @@ class WebExtensionPromptFeature(
         addon: Addon,
         promptRequest: WebExtensionPromptRequest.AfterInstallation.Permissions,
     ) {
-        if (hasExistingPermissionDialogFragment()) return
         showPermissionDialog(
-            addon,
-            promptRequest,
+            addon = addon,
+            onConfirm = promptRequest.onConfirm,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun handleOptionalPermissionsRequest(
+        addon: Addon,
+        promptRequest: WebExtensionPromptRequest.AfterInstallation.OptionalPermissions,
+    ) {
+        val shouldGrantWithoutPrompt = Addon.localizePermissions(promptRequest.permissions, context).isEmpty()
+
+        // If we don't have any promptable permissions, just proceed.
+        if (shouldGrantWithoutPrompt) {
+            promptRequest.onConfirm(true)
+            consumePromptRequest()
+            return
+        }
```

Comment:
Could you provide me more context around when this case could happen? As it seen a bit odd going the granting path instead of deny one, maybe I don't have enough context around it.

Reply:
Yep, that's what we are doing on Desktop: https://searchfox.org/mozilla-central/rev/29bdf6ff9965a647c6f64d63fed2b5bd094532c7/browser/modules/ExtensionsUI.sys.mjs#283-287 - Also, we're doing that in `AddonUpdater`: https://github.com/mozilla-mobile/firefox-android/blob/b0d480a03901c249f779bcc1c2f6cd0d38f645ae/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt#L214

Reply:
As this is an already known pattern, and we don't have nothing to do on the UI, wee could just add this logic to `GeckoEngine` and granting there, so the logic will be shared for all consumers.